src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -7560,7 +7560,14 @@ void M(int* p = stackalloc int[1])
             CreateStandardCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
                 // (4,21): error CS1525: Invalid expression term 'stackalloc'
                 //     void M(int* p = stackalloc int[1])
-                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "stackalloc").WithArguments("stackalloc").WithLocation(4, 21));
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "stackalloc").WithArguments("stackalloc").WithLocation(4, 21),
+                // (4,21): error CS0656: Missing compiler required member 'System.Span`1..ctor'
+                //     void M(int* p = stackalloc int[1])
```

Comment:
We shouldn't produce these errors if we cannot successfully bind to a span. if this is bound to a pointer (or not bound), the user shouldn't see such diagnostics?

Reply:
Since it is not in an initializer position it tries to assume Span type here, but span is missing. I can suppress the errors if we are not in a syntactically correct place.

---
In reply to: [139544034](https://github.com/dotnet/roslyn/pull/22162#discussion_r139544034) [](ancestors = 139544034)